gems/smithy/lib/smithy/anvil/client/views/shapes.rb

```diff
@@ -0,0 +1,238 @@
+            super()
+          end
+
+          attr_reader :shapes, :operation_shapes, :service_shape
+
+          def namespace
+            Tools::Namespace.namespace_from_gem_name(@plan.options[:gem_name])
+          end
+
+          def shapes_with_members
+            @shapes.reject { |s| s.members.empty? }
+          end
+
+          private
+
+          def assemble_shapes
+            serializable_shapes = []
+            operation_shapes = []
+            service_shape = nil
+
+            @model.shapes.each_value do |v|
+              case v.type
+              when 'service'
+                service_shape = assemble_service_shape(v)
+              when 'resource' then next
+              when 'operation'
+                operation_shapes << assemble_operation_shape(v)
+              else
+                serializable_shapes << assemble_shape(v)
+              end
+            end
+            [serializable_shapes, operation_shapes, service_shape]
+          end
+
+          def assemble_service_shape(shape_data)
+            ServiceShape.new(
+              id: shape_data.id,
+              name: shape_data.name,
+              traits: filter_traits(shape_data.traits),
+              version: shape_data.shape['version']
+            )
+          end
+
+          def assemble_operation_shape(shape_data)
+            OperationShape.new(
+              id: shape_data.id,
+              name: assemble_shape_name(shape_data.id),
+              input: assemble_shape_name(shape_data.shape['input']['target']),
+              output: assemble_shape_name(shape_data.shape['output']['target']),
+              errors: assemble_error_shapes(shape_data.shape['errors']),
+              traits: filter_traits(shape_data.traits)
+            )
+          end
+
+          def assemble_error_shapes(error_shapes)
+            return [] if error_shapes.nil?
+
+            error_shapes.each_with_object([]) do |err, a|
+              a << assemble_shape_name(err['target'])
+            end
+          end
+
+          def assemble_shape(shape_data)
+            SerializableShape.new(
+              id: shape_data.id,
+              name: shape_data.name,
+              shape_type: shape_type(shape_data.type),
+              traits: filter_traits(shape_data.traits),
+              members: assemble_member_shapes(shape_data.shape['members'])
+            )
+          end
+
+          def assemble_member_shapes(members)
+            return [] if members.nil?
+
+            members.each_with_object([]) do |(k, v), a|
+              a << MemberShape.new(
+                k.underscore,
+                assemble_shape_name(v['target']),
+                filter_member_traits(v['traits'])
+              )
+            end
+          end
+
+          def assemble_shape_name(shape_id)
+            if PRELUDE_SHAPES_MAP.include?(shape_id)
+              PRELUDE_SHAPES_MAP[shape_id]
+            else
+              shape_id.split('#').last
```

Comment:
This is the kind of stuff Vise is supposed to help with. This is the relative_id method right? Perhaps prelude shapes should be loaded into the model somehow.